src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,66 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * duke.Deadline is a duke.Task that accepts a task description and a deadline (dd-MM-yyyy)
+ */
+public class Deadline extends Task {
+
+    protected String by;
+    protected LocalDate byDate;
+
+
+    /**
+     * Constructor for duke.Deadline
+     *
+     * @param description description of the task
+     * @param by          date in the format of dd-MM-yyyy
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        this.byDate = LocalDate.parse(by, DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+    }
+
+
+    /**
+     * Private constructor just to convert text in save file to deadlines.
+     */
+    private Deadline(String description, LocalDate byDate, boolean isDone) {
+        super(description, isDone);
+        this.byDate = byDate; // this is retrieved in save file, so format is MM d yyyy
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString()
+            + "(by: " + byDate.format(DateTimeFormatter.ofPattern("MMM dd yyyy")) + ")";
+    }
+
+    /**
+     * returns a Deadline from a string with format "[D][X] Description (by: MMM d yyyy)"
+     * e.g. [D][ ] return book (by: Jun 6 2022)
+     *
+     * @param s String to convert into a Deadline
+     * @throws DukeException when the string doesn't start with deadline tag [D]
+     */
+    public static Deadline stringToDeadline(String s) throws DukeException {
+        if (!s.startsWith("[D][")) {
+            throw new DukeException("This string is not a duke.Deadline string!");
+        }
+        char isDoneString = s.charAt(4); //[T][X] checks if X is present
+        char X = 'X';
+        boolean isDone = isDoneString == X;
+        int idxOfBy = s.indexOf("(by:");
+
+        String description = s.substring(7, idxOfBy);
```

Comment:
Can 7 be declared as a constant? I think it will greatly improve the readability of your code!